src/code/z_bgcheck.c

```diff
@@ -4112,10 +4112,15 @@ s32 SurfaceType_IsIgnoredByProjectiles(CollisionContext* colCtx, CollisionPoly*
 }
 
 /**
- * CollisionPoly is conveyor enabled
- * Returns true if `poly` is a conveyor surface, else false
- */
-s32 SurfaceType_IsConveyor(CollisionContext* colCtx, CollisionPoly* poly, s32 bgId) {
+ * CollisionPoly is a floor conveyor
+ * 
+ * If the conveyor surface has a non-zero speed, then the conveyor will exhibit two types of behaviour depending on the return value:
+ * If the `poly` returns true, then it is a floor conveyor and will push player only while being stood on
+ * If the `poly` returns false, then it is a water conveyor and will push player only while being underwater
```

Comment:
we should probably say "while in water" as opposed to underwater. that makes it sound like you need to be diving under the water for the conveyor to work